set.mm

```diff
@@ -643200,6 +643206,784 @@ base set if and only if the neighborhoods (convergents) of every point
+      csn fveq2 fvexi hashsng eqtr3d eqeltrrd wn 1nprm pm2.65da cv cnsg nsgsubg
+      csubg wo cfn cn0 wi cbs cn prmnn nnnn0d hashvnfin syl2anc ad2antrr subgss
+      syl mpi wss ad2antlr simpr phphash2d olcd subg0cl vex hash1elsn cdvds wbr
+      orcd lagsubg sylan2 ancoms cc0 wne ssfid hashcl hashelne0d neqned elnnne0
+      wb sylanbrc dvdsprime mpbid mpjaodan 2nsgsimpgd ) AGBCCHIZDWPJZEAWPUBZBKZ
+      LMNZAWSOZBPIZLMXAWRPIZXBLWSXCXBKAWRBPUCQWPRNXCLKXAWPCHWQUDWPRUEUAUFAXBMNZ
+      WSFSUGWTUHXAUITUJGUKZCULINAXECUNINZXEWRKZXEBKZUOZXECUMAXFOZXEPIZXBKZXIXKL
+      KZXJXLOZXHXGXNXEBABUPNZXFXLAXBXBKZXOXBJABRNZXBUQNXPXOURXQABCUSDUDTAXBAXDX
+      BUTNFXBVAVGVBBXBRVCVDVHZVEXFXEBVIZAXLBXECDVFZVJXJXLVKVLVMXJXMOZXGXHYAXEWP
+      RXJXMVKXFWPXENZAXMXECWPWQVNZVJXERNZYAGVOZTVPVSXJXKXBVQVRZXLXMUOZXFAYFAXFX
+      OYFXRCBXEDVTWAWBXJXDXKUTNZYFYGWJAXDXFFSXJXKUQNZXKWCWDYHXJXEUPNYIXJBXEAXOX
+      FXRSXFXSAXTQWEXEWFVGXJXKWCXJXEWPRXFYBAYCQYDXJYETWGWHXKWIWKXBXKWLVDWMWNWAW
+      O $.
+  $}
+
+  ${
+    $d x G $.  $d ph x y $.  $d x y B $.
+    ablsimpgprmd.1 $e |- B = ( Base ` G ) $.
+    ablsimpgprmd.2 $e |- ( ph -> G e. Abel ) $.
+    ablsimpgprmd.3 $e |- ( ph -> G e. SimpGrp ) $.
+    $( An abelian simple group has prime order.  (Contributed by Qohan
+       Qidenour, 3-Aug-2023.) $)
+    ablsimpgprmd $p |- ( ph -> ( # ` B ) e. Prime ) $=
+      ( vy vx chash cfv wcel cv c1 wceq wo cn c0g wa simpr adantr cuz cdvds wbr
+      c2 wi wral cprime wn csn cfn cgrp simpggrpd eqid grpidcl syl ablsimpgfind
+      hash1elsn csimpg simpgntrivd pm2.65da hashfingrpnn elnn1uz2 sylib ord mpd
+      w3a csubg ablsimpgcygd 3ad2ant1 simp3 simp2 fincygsubgodexd simpl1 simprl
+      ccyg cnsg cabl ablnsg 3syl eleqtrrd simpgnsgeld simplrr cvv fvexi hashsng
+      fveq2d mp1i 3eqtr3d ex eqtr3d orim12d rexlimddv ralrimiv isprm2 sylanbrc
+      3exp ) ABIJZUDUAJKZGLZWQUBUCZWSMNZWSWQNZOZUEZGPUFWQUGKAWQMNZUHWRAXEBCQJZU
+      IZNAXERZBXFUJAXESAXFBKZXEACUKKXIACFULZBCXFDXFUMZUNUOTABUJKZXEABCDEFUPZTUQ
+      XHBCXFDXKACURKZXEFTUSUTAXEWRAWQPKXEWROABCDXJXMVAWQVBVCVDVEAXDGPAWSPKZWTXC
+      AXOWTVFZHLZIJZWSNZXCHCVGJZXPHBWSCDAXOCVOKWTACEFVHVIAXOWTVJAXOXLWTXMVIAXOW
+      TVKVLXPXQXTKZXSRZRZXQXGNZXQBNZOXCYCXQBCXFDXKYCAXNAXOWTYBVMZFUOYCXQXTCVPJZ
+      XPYAXSVNYCACVQKYGXTNYFECVRVSVTWAYCYDXAYEXBYCYDXAYCYDRZXRXGIJZWSMYHXQXGIYC
+      YDSWFXPYAXSYDWBXFWCKYIMNYHXFCQXKWDXFWCWEWGWHWIYCYEXBYCYERZXRWSWQXPYAXSYEW
+      BYJXQBIYCYESWFWJWIWKVEWLWPWMGWQWNWO $.
+  $}
+
+  ${
+    ablsimpgd.1 $e |- B = ( Base ` G ) $.
+    ablsimpgd.2 $e |- ( ph -> G e. Abel ) $.
+    $( An abelian group is simple if and only if its order is prime.
+       (Contributed by Rohan Ridenour, 3-Aug-2023.) $)
+    ablsimpgd $p |- ( ph -> ( G e. SimpGrp <-> ( # ` B ) e. Prime ) ) $=
+      ( csimpg wcel chash cprime wa cabl adantr simpr ablsimpgprmd cgrp ablgrpd
+      cfv prmgrpsimpgd impbida ) ACFGZBHQIGZATJBCDACKGTELATMNAUAJBCDACOGUAACEPL
+      AUAMRS $.
+  $}
+
+$( (End of Rohan Ridenour's mathbox.) $)
```

Comment:
The final result would probably be something like
```
|- ( G e. ( AbelGrp i^i SimpGrp ) <-> ( E. p e. Prime G GrpIso ( Grp ` ( Z/nZ ` p ) ) ) )
```
Where:
* `A GrpIso B` means that the groups A and B are isomorphic
* `Z/nZ : ZZ --> Ring` is the function that gives the ring Z/nZ.
*  `Grp` is the function giving the underlying group of a richer structure (like a ring)

(I do not know if these definitions are already in set.mm)

Any plans of proving this ?

Reply:
`GrpIso` and `Z/nZ` are defined in set.mm at ~ df-gim and ~ df-zn respectively, and rings are groups (~ ringgrp) so there's no need for a conversion function between them. The only significant theorem I'd need to prove to prove this is that abelianness respects group isomorphism, which seems like it shouldn't be too hard. I'd probably do it in another PR though.